Cwx.m

```diff
@@ -0,0 +1,360 @@
+//
+//  Cwx.m
+//
+//  Created by STU PHILLIPS on 7/7/15.
+//  Copyright (c) 2015 STU PHILLIPS. All rights reserved.
+//
+// LICENSE TERMS:
+// Stu Phillips, K6TU is the author and copyright of this software.
+// Copyright is assigned to Ridgelift, VC LLC.
+//
+// All rights are reserved.  Third parties may use this software under
+// the following terms:
+//
+// Educational, Non-commercial and Open Source use:
+// ------------------------------------------------
+// Any individual(s) or educational institutions may use this software at
+// no charge subject to the following conditions:
+// - K6TU Copyright is clearly acknowledged in the software
+//
+// If the software is developed other than for personal use and is distributed
+// in any form;
+// - Software incoporating the K6TU code is provided free of charge to end users
+// - Source code of the package/software including the K6TU code must be Open Source
+// - Source code of the package/software including the k6TU code must be publicly
+//   available on the Internet via github or similar repository system
+//
+// Commercial Use
+// --------------
+// The incorporation of the K6TU software in a proprietary product regardless of
+// whether the product is sold for a fee, bundled with another product at no cost
+// or in any use by a for-profit organization is expressly prohibited without a
+// specific license agreement from Stu Phillips, K6TU and Ridgelift VC, LLC.
+//
+// Violation of these Copyright terms will be protected by US & International law.
+//
+
+#import <Foundation/Foundation.h>
+#import "Cwx.h"
+#import "Slice.h"
+
+#define MAX_CWX_DELAY_MS 2000
+#define MAX_CWX_SPEED 100
+#define MAX_NUMBER_OF_MACROS 12
+#define MIN_CWX_SPEED 5
+
+
+@interface Cwx()
+
+// Radio object to which this Cwx belongs
+@property (strong, nonatomic, readwrite) Radio *radio;
+// Pointer to private run queue for Cwx
+@property (strong, nonatomic) dispatch_queue_t cwxRunQueue;
+// Array of strings
+@property (nonatomic, readwrite) NSMutableArray *macros;
+
+@property (strong, nonatomic) NSDictionary *cwxTokens;
+
+
+@end
+
+
+enum cwxToken {
+    breakinDelay=0,
```

Comment:
I usually add a cwxNullToken=0 as the first step so that all real tokens have a real value > 0